graphdatascience/procedure_surface/arrow/relationship_endpoints_helper.py

```diff
@@ -0,0 +1,142 @@
+
+        job_id = JobClient.run_job_and_wait(self._arrow_client, endpoint, config, show_progress)
+        summary = JobClient.get_summary(self._arrow_client, job_id)
+
+        if "configuration" in summary:
+            self._drop_write_internals(summary["configuration"])
+
+        return summary
+
+    def run_job_and_mutate(
+        self, endpoint: str, G: GraphV2, config: Dict[str, Any], mutate_property: str, mutate_relationship_type: str
+    ) -> Dict[str, Any]:
+        """Run a job, mutate node properties, and return summary with mutation result."""
+        show_progress = config.get("logProgress", True) and self._show_progress
+        job_id = JobClient.run_job_and_wait(self._arrow_client, endpoint, config, show_progress)
+        mutate_result = MutationClient.mutate_relationship_property(
+            self._arrow_client, job_id, mutate_relationship_type, mutate_property
+        )
+        computation_result = JobClient.get_summary(self._arrow_client, job_id)
+
+        # modify computation result to include mutation details
+        computation_result["relationshipsWritten"] = mutate_result.relationships_written
+        computation_result["mutateMillis"] = mutate_result.mutate_millis
+
+        if (config := computation_result.get("configuration", None)) is not None:
+            self._drop_write_internals(config)
+            config["mutateProperty"] = mutate_property
+            config["mutateRelationshipType"] = mutate_relationship_type
+
+        return computation_result
+
+    def run_job_and_stream(self, endpoint: str, G: GraphV2, config: Dict[str, Any]) -> DataFrame:
+        """Run a job and return streamed results."""
+        show_progress = config.get("logProgress", True) and self._show_progress
+        job_id = JobClient.run_job_and_wait(self._arrow_client, endpoint, config, show_progress=show_progress)
+        return JobClient.stream_results(self._arrow_client, G.name(), job_id)
+
+    def run_job_and_write(
+        self,
+        endpoint: str,
+        G: GraphV2,
+        config: Dict[str, Any],
+        *,
+        relationship_type_overwrite: str,
+        property_overwrites: Union[str, dict[str, str]],
+        write_concurrency: Optional[int],
+        concurrency: Optional[int],
+    ) -> Dict[str, Any]:
+        """Run a job, write results, and return summary with write time."""
+        show_progress = config.get("logProgress", True) and self._show_progress
+        job_id = JobClient.run_job_and_wait(self._arrow_client, endpoint, config, show_progress=show_progress)
+        computation_result = JobClient.get_summary(self._arrow_client, job_id)
+
+        if self._write_back_client is None:
+            raise Exception("Write back client is not initialized")
+
+        if isinstance(property_overwrites, str):
+            # The remote write back procedure allows specifying a single overwrite. The key is ignored.
+            property_overwrites = {property_overwrites: property_overwrites}
```

Comment:
Should we raise an error if more overwrites are given? 
In what case could there be more?

Reply:
v3 actually supports multiple property overwrites so this obsolete. for a single property we dont care around the key though and only fetch the value